domain/personallearningplan/src/main/kotlin/uk/gov/justice/digital/hmpps/domain/personallearningplan/service/GoalService.kt

```diff
@@ -35,6 +35,7 @@ class GoalService(
   private val goalEventService: GoalEventService,
   private val actionPlanPersistenceAdapter: ActionPlanPersistenceAdapter,
   private val actionPlanEventService: ActionPlanEventService,
+  private val goalNotesService: GoalNotesService,
```

Comment:
I'm really not sure about changing the behaviour of the domain GoalService to know how to get notes if we are saying that the Goal no longer has a note (as far as the domain is concerned), and that Note is it's own domain object.

Especially uncomfortable about this as the impl of GoalNotesService is defined in the application rather that the Notes domain 🤔 

I'm not sure this approach feels right

Reply:
As discussed - seems the lesser of a lot of evils.